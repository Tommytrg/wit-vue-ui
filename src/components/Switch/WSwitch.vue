<template>
  <label class="inline-flex items-center cursor-pointer">
    <input v-model="model" type="checkbox" value="" class="sr-only peer" />
    <div
      class="relative w-[44px] h-[24px] bg-white-200 peer-focus:outline-none peer-focus:ring-0 rounded-full peer dark:bg-black-500 peer-checked:after:translate-x-full rtl:peer-checked:after:-translate-x-full peer-checked:after:border-white-50 after:content-[''] after:absolute after:top-[2px] after:start-[2px] after:bg-white-50 after:border-white-50 after:border after:rounded-full after:h-[20px] after:w-[20px] after:transition-all dark:border-gray-700 peer-checked:bg-wit-blue-700 dark:peer-checked:bg-wit-blue-500"
    ></div>
  </label>
</template>

<script setup lang="ts">
const model = defineModel<boolean>({ required: true })
</script>
