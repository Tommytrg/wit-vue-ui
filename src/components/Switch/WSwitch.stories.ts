import type { Meta, StoryObj } from '@storybook/vue3'
import WSwitch from './WSwitch.vue'

const meta = {
  title: 'Example/WSwitch',
  component: WSwitch,
  tags: ['autodocs'],
  argTypes: {},
  args: {}
} satisfies Meta<typeof WSwitch>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
  render: (args) => ({
    components: { WSwitch },
    setup() {
      return { args }
    },
    template: `<WSwitch v-bind="args">Switch</Switch>`
  }),
  args: {
    modelValue: true
  }
}
