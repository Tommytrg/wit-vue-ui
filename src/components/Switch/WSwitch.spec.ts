import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import Switch from './WSwitch.vue'

describe('WSwitch', () => {
  it('it renders correctly', () => {
    const modelValue = true
    const wrapper = mount(Switch, { props: { modelValue } })

    expect(wrapper.element).toMatchSnapshot()
  })
})
