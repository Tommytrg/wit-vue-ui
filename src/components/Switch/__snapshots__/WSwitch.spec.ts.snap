// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WSwitch > it renders correctly 1`] = `
<label
  class="inline-flex items-center cursor-pointer"
>
  <input
    class="sr-only peer"
    type="checkbox"
    value=""
  />
  <div
    class="relative w-[44px] h-[24px] bg-white-200 peer-focus:outline-none peer-focus:ring-0 rounded-full peer dark:bg-black-500 peer-checked:after:translate-x-full rtl:peer-checked:after:-translate-x-full peer-checked:after:border-white-50 after:content-[''] after:absolute after:top-[2px] after:start-[2px] after:bg-white-50 after:border-white-50 after:border after:rounded-full after:h-[20px] after:w-[20px] after:transition-all dark:border-gray-700 peer-checked:bg-wit-blue-500 dark:peer-checked:bg-wit-blue-500"
  />
</label>
`;
